cpp/src/interop/from_arrow_host_strings.cu

```diff
@@ -0,0 +1,207 @@
+    if (schema->type == NANOARROW_TYPE_LARGE_STRING) {
+      return get_column_copy(&view, &offsets_array, data_type(type_id::INT64), true, stream, mr);
+    } else if (schema->type == NANOARROW_TYPE_STRING) {
+      return get_column_copy(&view, &offsets_array, data_type(type_id::INT32), true, stream, mr);
+    } else {
+      CUDF_FAIL("Unsupported string type", cudf::data_type_error);
+    }
+  }();
+  NANOARROW_THROW_NOT_OK(ArrowSchemaViewInit(&view, char_data_schema.get(), nullptr));
+
+  rmm::device_buffer chars(char_data_length, stream, mr);
+  CUDF_CUDA_TRY(cudaMemcpyAsync(chars.data(),
+                                reinterpret_cast<uint8_t const*>(char_array.buffers[1]),
+                                chars.size(),
+                                cudaMemcpyDefault,
+                                stream.value()));
+  auto const num_rows = offsets_column->size() - 1;
+  auto out_col        = make_strings_column(num_rows,
+                                     std::move(offsets_column),
+                                     std::move(chars),
+                                     input->null_count,
+                                     std::move(*mask.release()));
+
+  return input->offset == 0
+           ? std::move(out_col)
+           : std::make_unique<column>(
+               cudf::detail::slice(out_col->view(),
+                                   static_cast<size_type>(input->offset),
+                                   static_cast<size_type>(input->offset + input->length),
+                                   stream),
+               stream,
+               mr);
+}
+
+std::unique_ptr<column> from_arrow_stringview(ArrowSchemaView* schema,
+                                              ArrowArray const* input,
+                                              std::unique_ptr<rmm::device_buffer>&& mask,
+                                              rmm::cuda_stream_view stream,
+                                              rmm::device_async_resource_ref mr)
+{
+  ArrowArrayView view;
+  NANOARROW_THROW_NOT_OK(ArrowArrayViewInitFromSchema(&view, schema->schema, nullptr));
+  NANOARROW_THROW_NOT_OK(ArrowArrayViewSetArray(&view, input, nullptr));
+
+  // first copy stringview array to device
+  auto items   = view.buffer_views[1].data.as_binary_view;
+  auto d_items = rmm::device_uvector<ArrowBinaryView>(input->length, stream, mr);
+  CUDF_CUDA_TRY(cudaMemcpyAsync(d_items.data(),
+                                items + input->offset,
+                                input->length * sizeof(ArrowBinaryView),
+                                cudaMemcpyDefault,
+                                stream.value()));
+
+  // then copy variadic buffers to device
+  auto variadics     = std::vector<rmm::device_buffer>();
+  auto variadic_ptrs = std::vector<char const*>();
+  for (auto i = 0L; i < view.n_variadic_buffers; ++i) {
+    variadics.emplace_back(view.variadic_buffers[i], view.variadic_buffer_sizes[i], stream);
```

Comment:
There could be many of these buffers. What kind of performance are we getting here for the case of many small buffers? Do we want to reassemble some of this data on the host side (probably using multiple threads) before sending to device?

Reply:
Using the `ArrowArrayAppendString` did not fragment the buffers so this many small buffers case may be rare. 
I can follow up with a separate PR to handle that case if that is ok.